src/filters/ws.rs

```diff
@@ -324,6 +324,21 @@ impl Message {
         self.inner.is_pong()
     }
 
+    /// Try to get the close frame (close code and reason)
+    pub fn close_frame(&self) -> Result<Option<(u16, &str)>, ()> {
```

Comment:
Why return a `Result<Option>, ()>` instead of `Option`? `()` doesn't mean anything.

Reply:
To be consistent with [to_str()](https://github.com/seanmonstar/warp/blob/3bae3ecdad654e3c3145674a6387f2f1214e2ebe/src/filters/ws.rs#L342-L348)

Reply:
`to_str` returns only a `Result`, not `Result<Option>>`. I don't think that's a good API but I'll leave that to @seanmonstar to decide for sure.

Reply:
I don't remember why `to_str` does that... I'm not sure it's the best pattern to follow.

Just an `Option` here seems fine.

Reply:
Should both of the methods just return `Option` to be consistent? Or is just this one fine?